Group details screen (3)

OmarBasem opened the pull request:
Fixes #14475

This PR implements adding members and removing members from a group.

Designs: https://www.figma.com/file/7KIYbhoqNGAIFonE0w9TDz/Messages-for-Mobile?node-id=2105%3A522742&t=HU04xJbqoDL81ZPZ-0

OmarBasem committed:
group details screen 3

OmarBasem committed:
lint

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 63919608 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-14494/1/) | 2022-12-06 05:39:14 | ~2 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-221206-053702-639196-pr14494-tests.log) |

flexsurfer commented on src/quo2/components/avatars/group_avatar.cljs:
```diff
@@ -21,6 +21,6 @@
                 :justify-content  :center
                 :border-radius    (/ container-size 2)
-                :background-color (colors/custom-color-by-theme color 50 60)}
+                :background-color (colors/custom-hex-color color 50 60)}
```
could you please elaborate?

OmarBasem replied:
`custom-color-by-theme` is not working

Parveshdhull replied:
Please, also check quo2 preview while modifying quo2 components. This change is breaking the component

OmarBasem replied:
@Parveshdhull that's because quo2 preview is using color names, while actually the group color comes as a hex value

OmarBasem replied:
I think we need to give the preview component a list of hex values?

flexsurfer replied:
yeah i think the problem here is that we store this value in db i guess, and it's just regular hex value, so it seems like we can't use `custom-color-by-theme` or if its not dynamic, like we have a set of colors, we could bind them, idk what's better , but for now the simplest solution is to change preview component with hex values and use `colors/theme-alpha` instead of creating new `custom-hex-color` function wdyt ?

Parveshdhull replied:
We are going to get rid of old colors and designs, So I would recommend we can check if provided color is a keyword or not. And if not fallback to `:primary`
(Not sure, but we probably not going to support group theme customization for MVP? cc: @pedro-et).

Also using stored value will not be helpful, because those colors might not work with the current app background colors per theme, and changing the group avatar to cover this case of hard-coded color instead of dynamic, might create an unwanted chain reaction (changes in other places)

OmarBasem replied:
I do not think this is about group theme customization. The default group display photo is the group color. And I think it is always a hex value.

For now, I will provide some hex values for the Quo2 preview.

flexsurfer commented on src/status_im/ui2/screens/chat/group_details/view.cljs:
```diff
@@ -54,7 +53,13 @@
+
+(defn remove-members [chat-id]
+  (doseq [member-key @removed]
+    (rf/dispatch [:group-chats.ui/remove-member-pressed chat-id member-key true])))
```
better to have a new event for the list, and move `removed` atom to app-db

OmarBasem replied:
Currently, there is no status-go endpoint to remove a list of users. Opened an issue in status-go and will update it once we have that endpoint.

flexsurfer replied:
i mean re-frame app-db

ilmotta commented on src/status_im/ui2/screens/chat/group_details/view.cljs:
```diff
@@ -69,20 +74,21 @@
              [quo2/button {:style    {:flex 1}
-                           :on-press #(rf/dispatch [:bottom-sheet/hide])
-                           :disabled (zero? (count @added))}
+                           :on-press #(do
```
I think this is a good example where `fn` is preferable to using the function literal `#()`. I scanned all the lines inside the function to see if it was using any parameter from the anonymous function. If you use `(fn [] ...`), it immediately tells devs there's no argument to worry about.

Another reason to avoid this is that we're trying to follow the Clojure Style Guide (not because it's the one correct way, but because it helps everyone stay on the same page). See the section https://guide.clojure.style/#no-multiple-forms-fn-literals

OmarBasem replied:
Thanks for the explanation!

ilmotta commented on src/quo2/foundations/colors.cljs:
```diff
@@ -183,6 +183,23 @@
+    90 "E6"
+    "FF"))
+
+(defn custom-hex-color [color light-opacity dark-opacity]
```
Why can't we just use the `custom-color-by-theme` function? It seems to support different opacity levels based on the theme, which seems to be what you need, or not?

OmarBasem replied:
`custom-color-by-theme` is not working, not sure why

flexsurfer replied:
lets find out why it doesn't work, cc @Parveshdhull

Parveshdhull replied:
Hi @OmarBasem,
As I can see `custom-color-by-theme` is not working for you for group avatars, https://github.com/status-im/status-mobile/pull/14494/files#r1041791605
1. 50, 60 are color variants, not opacity
![image](https://user-images.githubusercontent.com/17097240/206179934-b2b6e864-d462-46ea-a3a8-113b9fce5139.png)
2. You need to provide color as `:purple` instead of hex value, then we will be able to find its appropriate variant, like :purple-50, :purple-60

OmarBasem replied:
Hi @Parveshdhull, How can I do that? The value comes as hex value

ilmotta replied:
@OmarBasem, are we going to be able to get rid of this new function or do you really need it? This is the only thing holding me from approving the PR, but if you think there's no way around this, I'm fine. If nothing else works, it would be nice to have a `TODO(OmarBasem)` saying something like *"custom-hex-color could be removed if XYZ is done"*.

OmarBasem replied:
@ilmotta I have added a todo: "custom-color-by-theme supports only string colors, so maybe we can get rid of it."

flexsurfer replied:
please use `theme-alpha` instead of creating new functions

qoqobolo commented:
Hey @OmarBasem , thank you for continuing to work on this feature!
Quick reminder: please add the `request-manual-qa` label once the PR is ready to be tested (if needed ofc).
Thanks!

OmarBasem commented:
> Hey @OmarBasem , thank you for continuing to work on this feature! Quick reminder: please add the `request-manual-qa` label once the PR is ready to be tested (if needed ofc). Thanks!

Sure @qoqobolo, thank you!

OmarBasem committed:
refactor

OmarBasem committed:
refactor

https://github.com/status-im/status-go/compare/684e9654...b4bdfd3d

OmarBasem committed:
refactor

https://github.com/status-im/status-go/compare/684e9654...b4bdfd3d

OmarBasem committed:
refactor

https://github.com/status-im/status-go/compare/684e9654...b4bdfd3d

OmarBasem committed:
refactor

ilmotta commented on src/status_im2/subs/contact.cljs:
```diff
@@ -84,6 +84,26 @@
+    (map #(first (groups %)) (distinct (map f coll)))))
+
+(re-frame/reg-sub
+ :contacts/add-members-sections
```
@OmarBasem, could you add some tests to this subscription? This is considered a layer-3 sub, we should test them, especially a non-trivial like this one. I've recently merged a [PR](https://github.com/status-im/status-mobile/pull/14472) showing how to do it, it's almost as simple as testing any other pure function.

Test examples for layer-3 subs:

- https://github.com/status-im/status-mobile/blob/develop/src/status_im2/subs/communities_test.cljs
- https://github.com/status-im/status-mobile/blob/develop/src/status_im2/subs/wallet/wallet_test.cljs
- https://github.com/status-im/status-mobile/blob/develop/src/status_im2/subs/activity_center_test.cljs

OmarBasem replied:
Sure @ilmotta, thank you!

OmarBasem replied:
@ilmotta I am currently unable to make the test pass successfully. I do not want to spend more time on this right now, so I have opened an issue for it (https://github.com/status-im/status-mobile/issues/14531) and will get back to it later (if you don't mind).

ilmotta replied:
It's challenging I know, not the simplest of the tests to start, given its stateful nature. Thanks for trying, perhaps let's pair program on that new issue next week.

OmarBasem replied:
Thanks @ilmotta, we can try again with it sometime next week.

OmarBasem committed:
update status-go version

https://github.com/status-im/status-go/compare/684e9654...b4bdfd3d

OmarBasem committed:
refactor

OmarBasem committed:
tests

OmarBasem committed:
tests

OmarBasem committed:
refactor

OmarBasem committed:
refactor

ibrkhalil commented on src/status_im/group_chats/core.cljs:
```diff
@@ -196,11 +215,16 @@
 
-(fx/defn add-participants-toggle-list
-  {:events [:group/add-participants-toggle-list]}
+(fx/defn reset-add-participants
```
Since we're picking a new name
Will `clear-added-participants` be better ?

ibrkhalil commented on src/status_im/group_chats/core.cljs:
```diff
@@ -196,11 +215,16 @@
   [{db :db}]
   {:db (assoc db :selected-participants #{})})
 
+(fx/defn reset-remove-members
```
`clear-removed-members` ?

ibrkhalil commented on src/status_im/ui2/screens/chat/group_details/view.cljs:
```diff
@@ -51,38 +49,33 @@
+                                   (rf/dispatch [:group-chats.ui/add-members-pressed])
+                                   (rf/dispatch [:group-chats.ui/remove-members-pressed])
+                                   (rf/dispatch [:bottom-sheet/hide]))
+                       :disabled (and (zero? (count selected-participants)) (zero? (count deselected-members)))}
```
I think the Clojure way of checking if a sequence is not empty is by using `seq`
https://clojuredocs.org/clojure.core/seq
Example:
```clj
(seq [])
nil
(seq [1])
(1)
(seq #{})
nil
(seq #{1})
(1)

```

ibrkhalil commented on src/status_im2/common/contact_list_item/view.cljs:
```diff
@@ -18,19 +18,27 @@
-                      :on-change        (fn [selected] (if selected
-                                                         (swap! added conj public-key)
-                                                         (reset! added (remove #(= % public-key) @added))))}])]))
+                      :disabled?        (and member? (not admin?))
```
Just a suggestion
Can we move this to the `let` binding?
```clj
(and member? (not admin?))
```

OmarBasem replied:
I think it's fine that way. Also, can't think of a meaningful variable name for it.

ibrkhalil reviewed (approved):
Non blocking comments
I second Icaro on writing tests for `:contacts/add-members-sections`

status-im-auto commented:
## 100% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 0
Passed tests: 8
Not executed tests: 2
```
```
IDs of not executed tests: 702807,702808 
```
<h3>Not executed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14494.apk&PR_ID=14494&TR_CASE_IDS=702807,702808">Rerun not executed tests</a></li></details><h3>Passed tests (8)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067876">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1ce5ef64ae004d988f62a2cfdb18906a?auth=4b7270f15d8835ae486de2ad9d8db32a#272">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3ea9f0a348e04df9ace3f3e78f9d5997?auth=0dc31195dafb3fbd9f66e20d2ba64839#173">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067877">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1ce5ef64ae004d988f62a2cfdb18906a?auth=4b7270f15d8835ae486de2ad9d8db32a#193">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3ea9f0a348e04df9ace3f3e78f9d5997?auth=0dc31195dafb3fbd9f66e20d2ba64839#108">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067873">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1ce5ef64ae004d988f62a2cfdb18906a?auth=4b7270f15d8835ae486de2ad9d8db32a#83">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3ea9f0a348e04df9ace3f3e78f9d5997?auth=0dc31195dafb3fbd9f66e20d2ba64839#57">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067874">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/1ce5ef64ae004d988f62a2cfdb18906a?auth=4b7270f15d8835ae486de2ad9d8db32a#103">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3ea9f0a348e04df9ace3f3e78f9d5997?auth=0dc31195dafb3fbd9f66e20d2ba64839#78">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMediumMultipleDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067875">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/33b5e9e701ed49b39e1fcd015c63445d?auth=d9a3f55a3e5e49de04bdaabcfe4fbec8#64">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3d7c3cba4e904e038ecdba0c00f8cc74?auth=3cb12b3329390d4d3d16d3caecee7246#59">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeeplinkOneDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067880">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7019cfc9d404e8d84b6efd7633aed4f?auth=7d26b9969710c1a36921352397a1d139#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067878">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7019cfc9d404e8d84b6efd7633aed4f?auth=7d26b9969710c1a36921352397a1d139#40">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3067879">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b7019cfc9d404e8d84b6efd7633aed4f?auth=7d26b9969710c1a36921352397a1d139#97">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

ilmotta commented on src/status_im/group_chats/core.cljs:
```diff
@@ -41,6 +41,15 @@
+  {::json-rpc/call [{:method      "wakuext_removeMembersFromGroupChat"
+                     :params      [nil current-chat-id deselected-members]
+                     :js-response true
+                     :on-success  #(re-frame/dispatch [:chat-updated % true])}]})
```
@OmarBasem, the `json-rpc/call` effect unfortunately has no check for the existence of the `:on-error` option, which means if you don't pass an empty anonymous function to `:on-error` this event handler will explode if an error happens.

So unfortunately, for now, we need to add `:on-error #()`. Another one of those things that could be caught by a test.

ilmotta commented on src/status_im/group_chats/core.cljs:
```diff
@@ -41,6 +41,15 @@
                      :on-success  #(re-frame/dispatch [:chat-updated % do-not-navigate?])}]})
 
+(fx/defn remove-members
+  "remove members from a group chat"
```
This docstring seems quite redundant.

ilmotta commented on src/status_im/group_chats/core.cljs:
```diff
@@ -176,6 +185,16 @@
        (= constants/invitation-state-removed)))
 
+(fx/defn deselect-member
+  {:events [:deselect-member]}
```
Perhaps the new guidelines about [Event names](https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#subscription-names-and-event-names) are not clear enough, but we almost never want to register event handlers without a qualification in the keyword.

In this case, it could be registered as `:group-chats/deselect-member`. Similarly for `:group-chats/undo-deselect-member`, but never just `:undo-deselect-member`.

The idea being that we only want events registered without qualification if they're highly generic, reusable, but anything domain related should be qualified.

And now that I look at the event name keywords in this namespace, they are not consistent. Some don't have qualification, some use `:group`, others use `:group-chats`, etc. Not saying your PR should fix this, but it'd be great to have this improved in future PRs.

OmarBasem replied:
Yes I was confused what should I name it exactly due to the inconsistency. I have opened an issue for it: https://github.com/status-im/status-mobile/issues/14547

ilmotta commented on src/status_im/ui2/screens/chat/group_details/view.cljs:
```diff
@@ -51,38 +49,33 @@
+   (fn []
+     (let [{window-height :height} (rn/use-window-dimensions)
+           safe-area             (safe-area/use-safe-area)
+           selected-participants (rf/sub [:selected-participants])
```
We should avoid at all costs registering global subscriptions. They will be poorly cached by re-frame, i.e. every db/state change will cause the recomputation of top level subscriptions.

We should also register subscriptions with qualification, never global names like `:selected-participants`. See https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#subscription-names-and-event-names

ilmotta commented on src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs:
```diff
@@ -22,7 +22,7 @@
                      {:key   c
                       :value c})
-                   (keys colors/customization))}])
+                   ["#ff0000" "#0000ff"])}])
```
Could we avoid the hardcoded string colors here?

OmarBasem replied:
This is for the preview screen. And it is temporary only.

Parveshdhull replied:
Please add todo

J-Son89 replied:
@OmarBasem, @Parveshdhull can we start just creating issues in GitHub instead. The issues can just point to where in the code the issue exists 👍

OmarBasem replied:
There is an issue @J-Son89 on GitHub and linked in the todo

ilmotta commented on src/status_im2/subs/contact.cljs:
```diff
@@ -84,6 +84,26 @@
        sort
        vals)))
 
+(defn distinct-by [f coll]
```
To me, this kind of function should be moved to `utils/core.cljc` and it should be unit tested.

ilmotta commented on src/status_im2/subs/root.cljs:
```diff
@@ -97,6 +97,7 @@
 (reg-root-key-sub :group-chat-profile/editing? :group-chat-profile/editing?)
 (reg-root-key-sub :group-chat-profile/profile :group-chat-profile/profile)
 (reg-root-key-sub :selected-participants :selected-participants)
+(reg-root-key-sub :deselected-members :deselected-members)
```
As I mentioned earlier in this PR, we should generally avoid root state like this due to performance issues, but also because it's difficult to reason about it. For instance, "deselected members of what?".

Unfortunately the current app DB is badly designed in many ways, so I understand you're kind of just going with the grain.

A better design IMO would group related subscriptions under a common bounded context (e.g. chat or group chat, etc). This is really important, perhaps we should document it somehow.

flexsurfer commented on shadow-cljs.edn:
```diff
@@ -83,6 +83,7 @@
            :optimizations    :simple
            :target           :node-test
            :main             status-im.test-runner/main
+           :ns-regexp        "status-im2.subs.contact-test"
```
don't forget to remove this before merge

pavloburykh commented:
@OmarBasem please, ping me up when all review comments will be resolved and PR will be ready for manual testing. As I suppose there will be some additional commits. Thanx.

Parveshdhull commented on src/quo2/foundations/colors.cljs:
```diff
@@ -183,6 +183,24 @@
+(defn custom-hex-color [color light-opacity dark-opacity]
+  (str color (opacity->hex (if (theme/dark?) dark-opacity light-opacity))))
+
+;; TODO (OmarBasem): custom-color-by-theme supports only string colors, so maybe we can get rid of it.
```
Hi @OmarBasem,
Comment is at the wrong place, [suggestion](https://github.com/status-im/status-mobile/pull/14494/files#r1047095653) is about later removing `custom-hex-color` not `custom-color-by-theme`.
Groups and other parts of the app is using colors based on name/string.

I have no idea why even we are [using](https://github.com/status-im/status-mobile/pull/14494#discussion_r1040496455) `(colors/custom-hex-color color 50 60)`.
As I [mentioned](https://github.com/status-im/status-mobile/pull/14494/files#r1042148417) above 50 & 60 are not opacity they are just color variants.
And also currently we are getting the group avatar color as hex, but that's only for now.
Either we can just follow our approach of custom-color-by-them and hard-code (or fallback) primary color for now, or if we want to use the hex color provided by the group, we should directly use that. Instead of changing opacity.

OmarBasem replied:
Hi @Parveshdhull, Alright I understand, but the 50 and 60 in the group avatar are actually opacity, that's how they are in the designs. And the group colors comes as hex, that's why we need this method. Since `custom-color-by-theme` is used by other components, then I guess we need both methods, right?

Parveshdhull replied:
>  but the 50 and 60 in the group avatar are actually opacity

they are not opacity - https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=179%3A9760&t=XVp7uXCCPclgNDXg-0

![1](https://user-images.githubusercontent.com/17097240/207587318-94bfaa08-f328-43c4-98b3-001bd03158b2.png)
![2](https://user-images.githubusercontent.com/17097240/207587332-13a8ceb0-b44d-43ff-ae05-4f3210aaa788.png)
![3](https://user-images.githubusercontent.com/17097240/207587335-23852932-61f1-4e81-8203-3416ccc47b14.png)

OmarBasem replied:
@Parveshdhull you are right. My bad. Sorry! I will fix it.

OmarBasem replied:
@Parveshdhull Since the colors come as hex from the backend, and the designs are using specific customizations of the (string) colors, either the designs need to be updated, or the way colors are represented needs to change. So for now I think the best option is to let the `group-avatar` use the color as it is.

OmarBasem commented:
> @OmarBasem please, ping me up when all review comments will be resolved and PR will be ready for manual testing. As I suppose there will be some additional commits. Thanx.

Hi @pavloburykh, PR is ready to be QA tested. Current discussions are on some technicalities. The implementation is complete.

pavloburykh commented:
@OmarBasem Thanx for the PR. Please, take a look at the following issues

### ISSUE 1 Owner is unable to add new group members but regular member does

**Steps:** 

1. Create a group by User A
2. Add User B as a member during group creation
3. Open group details - Add member by User A an User B
4. Try to add new member (User C) by owner User A
5. Try to add new member (User C) by regular member User B
6. Observe the result

**Actual result:** group owner User A can put checkmark for User C in the list, but Save button does not appear. At the same time regular User B can put checkmark for User C in the list and save changes. 

**Expected result:** owner User A should be able to add new group members, regular members should not possess this right. 

Video of group owner screen:

https://user-images.githubusercontent.com/97245802/207601906-dfcb0957-78f1-4216-9b73-ec58eff0c657.mp4

Video of regular member screen:


https://user-images.githubusercontent.com/97245802/207602019-8a5b4f2e-730b-43ec-beb6-10bfb7b620a4.mp4

pavloburykh commented:
### QUSETION 1: Why should we show Add member option for regular group members, who should not possess right to add new members? 

![photo_2022-12-14 15 04 53](https://user-images.githubusercontent.com/97245802/207602822-ff487757-16f4-403f-bfb8-b9d5b421d1e2.jpeg)

pavloburykh commented:
### ISSUE 2 It is possible to add a new group member who is not mutual contact

**Prerequisites** User A and User B are NOT mutual contacts (e.g. User A has User B in contact list but User B has not User A in his contact list) 
**Steps:** 

1. Create a new group be User A
2. Open group details - Add member
3. Add User B as new member

**Actual result:** User B is added as a new member.

**Expected result:** it should not be possible to add User B as a new member as User A and User B are not mutual contacts.  Probably we should show some pop up with explanation in this case. We should ask design team what kind of pop up and what kind of text should be.

pavloburykh commented:
### ISSUE 3 Group members are always listed as "offline" even if they are online

**Steps:**

1. Create a group by User A
2. Add User B as a member during group creation
3. Stay logged in by User B
4. Open group details by User A
5. Pay attention at the list of offline members

**Actual result:** Members are listed in the offline section regardless their actual state

![photo_2022-12-14 15 18 18](https://user-images.githubusercontent.com/97245802/207605702-039ef80f-e327-4a7a-9c96-43c47f6aeeda.jpeg)

pavloburykh commented:
### ISSUE 4: Accessibility ids are missing for the following elements

1. Pinned messages, Mute group, Manage members elements:
<img width="422" alt="Appium 2022-12-14 17-17-34" src="https://user-images.githubusercontent.com/97245802/207635199-0b40717b-dc39-4349-80c2-2e4f815d4509.png">

2. Counters for Pinned messages and Members 

<img width="420" alt="Appium 2022-12-14 17-19-19" src="https://user-images.githubusercontent.com/97245802/207635491-a17bffa3-2a29-4112-a529-9e99dd77f5b4.png">

3. Group details menu button

<img width="1244" alt="Appium 2022-12-14 17-54-44" src="https://user-images.githubusercontent.com/97245802/207645329-466114f5-e538-4a5e-8a14-a5552c08e108.png">

4. Close button on Add member screen

<img width="1227" alt="Appium 2022-12-14 18-04-04" src="https://user-images.githubusercontent.com/97245802/207646417-e93d633d-de8f-41f4-ab94-c0370b44ba08.png">

5. Member elements in the list of Add member screen

<img width="1197" alt="Appium 2022-12-14 18-06-05" src="https://user-images.githubusercontent.com/97245802/207648876-38f307d0-382e-4ca4-ae54-100f7cbcefa9.png">

6. Checkbox on Add member screen

<img width="1239" alt="Appium 2022-12-14 18-05-40" src="https://user-images.githubusercontent.com/97245802/207648987-d6a02bf6-613a-4b61-9b33-0b326d2fccb8.png">

7. Save button on Add member screen

![photo_2022-12-14 18 16 12](https://user-images.githubusercontent.com/97245802/207649610-173a470e-f228-47bb-a393-5ead4f97a16e.jpeg)

pavloburykh commented:
### ISSUE 5: Sometimes member is not removed from group after unchecking him on Manage member screen 

Bug is not reproducible every time. 

**Steps:**

1. Add new member to the group
2. Try to remove group member by unchecking him in the list of Manage members screen and Saving changes
3. See if member is removed 

**Actual result:** sometimes member is not removed from the first time (see Video below). 

**Expected result:** member is removed from the group


https://user-images.githubusercontent.com/97245802/207883143-cd7f6b7d-3141-4e49-8082-fcc6614ebb39.mp4

OmarBasem committed:
refactor

https://github.com/status-im/status-go/compare/684e9654...b4bdfd3d

OmarBasem committed:
lint

https://github.com/status-im/status-go/compare/684e9654...b4bdfd3d

OmarBasem commented:
@pavloburykh thank you for reviews! I have fixed the issues. Please let me know if there are more issues.

Issue 1: that's a problem with the `bottom-sheet` component. Not part of this PR.
Question 1: Yes, regular members can add new members.
Question 2: Yes, there is difference in functionality. Regualar members can only add members. Admins can add and remove members.
Question 3: Probably there should be. According to the design it's not on this screen.
Issue 2: Fixed it.
Issue 3: Currently "offline" is hard coded. Functionality needs to be implemented on the backend on `status-go`.
Issue 4: Fixed it.
Issue 5: Fixed it.

Parveshdhull commented:
> Issue 3: Currently "offline" is hard coded. Functionality needs to be implemented on the backend on status-go.

you can use `(rf/sub [:visibility-status-updates/online? contact-id])`

Parveshdhull commented on src/quo2/components/avatars/group_avatar.cljs:
```diff
@@ -21,6 +21,6 @@
                 :justify-content  :center
                 :border-radius    (/ container-size 2)
-                :background-color (colors/custom-color-by-theme color 50 60)}
+                :background-color color}
```
Please don't remove this line, only comment it out.
Also will be better to add a todo, because this is only temporary, we eventually need to support custom colors.

Parveshdhull commented on src/status_im/group_chats/core.cljs:
```diff
@@ -83,7 +92,9 @@
                      :js-response true
-                     :on-success  #(re-frame/dispatch [:chat-updated %])}]})
+                     :on-success  #(do
+                                     (println "SUCCESSADDMEMBERS" selected-participants)
```
leftover?

Parveshdhull commented on src/status_im/ui2/screens/chat/group_details/view.cljs:
```diff
@@ -51,38 +50,35 @@
+         [quo2/button {:style    {:flex 1}
+                       :accessibility-label :save
+                       :on-press (fn []
+                                   (rf/dispatch [:group-chats.ui/add-members-pressed])
+                                   (rf/dispatch [:group-chats.ui/remove-members-pressed])
+                                   (rf/dispatch [:bottom-sheet/hide]))
```
Please can you elaborate, why add and remove events simultaneously?

OmarBasem replied:
An admin can select new contacts to add to group, and deselect current members to remove from group. Save button will do both updates.

pavloburykh commented:
> 📱`ipa`

Thanx @OmarBasem !

Problem is that ISSUE 1 is blocker for testing PR on Android. So I guess we need to fix the problem bottom-sheet component first.

OmarBasem commented:
> ["#ff0000" "#0000ff"]

Thanks! good to know

OmarBasem commented:
> > Issue 3: Currently "offline" is hard coded. Functionality needs to be implemented on the backend on status-go.
> 
> you can use `(rf/sub [:visibility-status-updates/online? public-key])`

Thanks! good to know

OmarBasem committed:
qa fixes

OmarBasem commented:
Thanks @pavloburykh, PR is ready to be tested.

Parveshdhull reviewed (approved):
Thank you for changes, lgtm.

pavloburykh commented:
> @pavloburykh thank you for reviews! I have fixed the issues. Please let me know if there are more issues.
> 
> Issue 1: that's a problem with the `bottom-sheet` component. Not part of this PR. Question 1: Yes, regular members can add new members. Question 2: Yes, there is difference in functionality. Regualar members can only add members. Admins can add and remove members. Question 3: Probably there should be. According to the design it's not on this screen. Issue 2: Fixed it. Issue 3: Currently "offline" is hard coded. Functionality needs to be implemented on the backend on `status-go`. Issue 4: Fixed it. Issue 5: Fixed it.

@OmarBasem thanx for the fixes!

Still reproducing [ISSUE 5](https://github.com/status-im/status-mobile/pull/14494#issuecomment-1353162295)


Uploading telegram-cloud-document-2-5251633362392589333.mp4…

pavloburykh commented:
### ISSUE 5 Removed group member still has an active input field and can send messages which are not visible for group members

**Steps:** 

1. Remove group member via Manage members functionality
2. See if removed group member still has an active input field in the group and can send messages

**Actual result:** removed group member has active input field and can send messages though messages are not delivered to the group

**Expected result:** remove group member should not have an active input field in the group. 


https://user-images.githubusercontent.com/97245802/208147508-71d3e88f-02e6-4832-aaf0-bbdd76ee61df.mp4

OmarBasem committed:
refactor

OmarBasem committed:
rebase

pavloburykh commented:
Hey @OmarBasem Please, ping me up when PR will be ready for re-test. Thanx!

OmarBasem committed:
qa fixes

https://github.com/status-im/status-go/compare/684e9654...b4bdfd3d

OmarBasem committed:
lint

OmarBasem commented:
PR is ready to be tested.
@pavloburykh @VladimrLitvinenko

OmarBasem committed:
status-go version

https://github.com/status-im/status-go/compare/684e9654...b4bdfd3d

OmarBasem committed:
lint

OmarBasem committed:
refactor

OmarBasem commented:
@flexsurfer please let me know if there are any more issues with the PR

VladimrLitvinenko commented:
> PR is ready to be re-tested. @pavloburykh @VladimrLitvinenko

Thank you. Soon it'll be tested

flexsurfer commented on src/quo2/components/avatars/group_avatar.cljs:
```diff
@@ -21,6 +21,7 @@
                 :justify-content  :center
                 :border-radius    (/ container-size 2)
-                :background-color (colors/custom-color-by-theme color 50 60)}
+                ;:background-color (colors/custom-color-by-theme color 50 60)
```
could you elaborate?

Parveshdhull replied:
Commenting out without todo doesn't look much good. But this was done due to this: https://github.com/status-im/status-mobile/pull/14494#discussion_r1050603672

Maybe there is a better way?

OmarBasem replied:
@flexsurfer I had the todo in the quo_preview file. I added it to this file as well.

flexsurfer commented on src/status_im/group_chats/core.cljs:
```diff
@@ -79,11 +88,11 @@
+  [{{:keys [current-chat-id]} :db :as cofx}]
   {::json-rpc/call [{:method      "wakuext_addMembersToGroupChat"
-                     :params      [nil current-chat-id selected-participants]
+                     :params      [nil current-chat-id (get-in cofx [:db :group-chat/selected-participants])]
```
you could use namespaced keys destructuring

flexsurfer replied:
`:group-chat/keys [selected-participants]`

OmarBasem replied:
Thank you. Fixed.

flexsurfer commented on src/status_im2/contexts/chat/messages/view.cljs:
```diff
@@ -58,7 +58,8 @@
 (defn chat-render []
   (let [;;we want to react only on these fields, do not use full chat map here
-        {:keys [chat-id show-input?] :as chat} (rf/sub [:chats/current-chat-chat-view])
+        show-input? (not (rf/sub [:group-chat/removed-from-current-chat?]))
```
this change looks suspicious , could you elaborate?

OmarBasem replied:
We hide the input when a member has been removed. `show-input` value from `:chats/current-chat-chat-view` was always returning true even when the member is removed. So I used this other subscription `:group-chat/removed-from-current-chat?]`

flexsurfer commented on src/status_im2/setup/db.cljs:
```diff
@@ -11,7 +11,8 @@
              :chats                              {}
              :current-chat-id                    nil
-             :selected-participants              #{}
+             :group-chat/selected-participants              #{}
```
fromatting

OmarBasem committed:
refactor

VladimrLitvinenko commented:
@OmarBasem Thanx for PR. Some additional 

**ISSUE 7: The offline users are shown with green badge**

**Preconditions**
Set contacts for the user or use the following:
(seed phrase: spike spoil hidden sad juice force trap bind situate consider suit demand)

**Steps to reproduce:**
1. Create the group chat -> invite several users during group chat creation
2. Go to the 'Group details' page

**Actual result:**
Offline users are shown with green badge
![image](https://user-images.githubusercontent.com/52490791/208653058-1fbb6daf-57cd-4bde-bbcf-04eb1b43115c.png)

**Expected result:**
Offline users are shown with grey badge
![image](https://user-images.githubusercontent.com/52490791/208652640-29edf6e0-2c92-4120-84d9-924c98c2b043.png)

**ISSUE 8: The last user is not visible on the 'add members' page**
**Preconditions**
Set 10 contacts for the user or use the following:
(seed phrase: spike spoil hidden sad juice force trap bind situate consider suit demand)

**Steps to reproduce:**
1. Create group chat
4. Go to the details page of group chat
5. Tap 'Manage members'
6. Scroll to the last user

**Actual result:**
Last user is not visible fully and overlapped by the [save] button

https://user-images.githubusercontent.com/52490791/208651813-46758aa7-0ab6-4764-a791-9df1262a1741.mp4

**Expected result:**
All users are visible including last one

**ISSUE 9. Some users are not saved as invited into the group after theirs selection and saving**
**Steps to reproduce**
1. Use user that has at least 10 contacts
2. Create the group chat
3. Go to the 'Group details' page
4. Tap 'manage members'
5. Try to add and remove some contacts
6. Tap [save] button

**Actual result:**
The users are not saved as members
https://user-images.githubusercontent.com/52490791/208676236-083def0f-de93-46a1-9866-c2bfe646043e.mp4

**Expected result:**
The selected users are saved

**Additional result:**
Can't catch for which particular users it happens. Maybe you can see why this issue happens in the logs

**Logs:** 
[geth.log](https://github.com/status-im/status-mobile/files/10268672/geth.log)


**Question 4: How the user which is removed from the group should see the 'add members' page? Currently he is able to go to this page and checkmarks other users. As for me, the chat group should disappear for him**

VladimrLitvinenko commented:
@OmarBasem **Question 4: How the user which is removed from the group should see the 'add members' page? Currently, he is able to go to this page and checkmark other users. As for me, the chat group should disappear for him**

OmarBasem committed:
rebase

OmarBasem committed:
qa fixes

OmarBasem commented:
@VladimrLitvinenko thanks for your reviews! I have fixed the issues apart from issue 5 which I am not able to reproduce effectively. I have opened an issue for it https://github.com/status-im/status-mobile/issues/14613
Issue 7: fixed
Issue 8: fixed
Question 4: fixed. Group details option no longer displayed if a user is removed.

OmarBasem committed:
rebase

OmarBasem committed:
rebase

status-im-auto commented:
## 100% of end-end tests have passed
```
Total executed tests: 18
Failed tests: 0
Passed tests: 18
Not executed tests: 2
```
```
IDs of not executed tests: 702844,702847 
```
<h3>Not executed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&APK_NAME=14494.apk&PR_ID=14494&TR_CASE_IDS=702844,702847">Rerun not executed tests</a></li></details><h3>Passed tests (18)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestGroupChatMediumMultipleDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068790">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/b6864df65c7e4bb5852817c6e607f660?auth=f755fb265f313ff89cc56487357769f1#65">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/bbcc975a9c9347d181b90a62c1c22ef8?auth=50da2b2f96b5ac9a28aeb68fb45f6670#60">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068806">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/48a2510ab7b04f72af83fa62aee4b8b5?auth=385e5fce23a2b9763b86b1312ea820ec#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068791">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3c72c854b0694c4cbe6436e520cbe749?auth=28a4f944a55d169c6c346fa7a662388d#273">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d082f0d8a7144642b8e5c69b45a566a2?auth=ac61c34707a1d5b9f6cfd0d3c979177b#174">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068792">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3c72c854b0694c4cbe6436e520cbe749?auth=28a4f944a55d169c6c346fa7a662388d#194">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d082f0d8a7144642b8e5c69b45a566a2?auth=ac61c34707a1d5b9f6cfd0d3c979177b#109">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068788">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3c72c854b0694c4cbe6436e520cbe749?auth=28a4f944a55d169c6c346fa7a662388d#84">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d082f0d8a7144642b8e5c69b45a566a2?auth=ac61c34707a1d5b9f6cfd0d3c979177b#58">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068789">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3c72c854b0694c4cbe6436e520cbe749?auth=28a4f944a55d169c6c346fa7a662388d#104">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d082f0d8a7144642b8e5c69b45a566a2?auth=ac61c34707a1d5b9f6cfd0d3c979177b#79">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068800">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/af898382715e49cf8db0b8963ecca555?auth=f106daa8845d3efd69267ee51256afc4#181">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdcaf6f5623f4b0d84cfc45cbf81356e?auth=e946fd4077294408c08e31076a5535d4#112">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068805">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/af898382715e49cf8db0b8963ecca555?auth=f106daa8845d3efd69267ee51256afc4#246">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdcaf6f5623f4b0d84cfc45cbf81356e?auth=e946fd4077294408c08e31076a5535d4#141">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068801">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/af898382715e49cf8db0b8963ecca555?auth=f106daa8845d3efd69267ee51256afc4#196">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdcaf6f5623f4b0d84cfc45cbf81356e?auth=e946fd4077294408c08e31076a5535d4#125">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068799">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/af898382715e49cf8db0b8963ecca555?auth=f106daa8845d3efd69267ee51256afc4#161">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdcaf6f5623f4b0d84cfc45cbf81356e?auth=e946fd4077294408c08e31076a5535d4#83">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068802">test_community_mark_all_messages_as_read</a>, id: 702842 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/af898382715e49cf8db0b8963ecca555?auth=f106daa8845d3efd69267ee51256afc4#226">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdcaf6f5623f4b0d84cfc45cbf81356e?auth=e946fd4077294408c08e31076a5535d4#133">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068798">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/af898382715e49cf8db0b8963ecca555?auth=f106daa8845d3efd69267ee51256afc4#133">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdcaf6f5623f4b0d84cfc45cbf81356e?auth=e946fd4077294408c08e31076a5535d4#64">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068803">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/af898382715e49cf8db0b8963ecca555?auth=f106daa8845d3efd69267ee51256afc4#146">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cdcaf6f5623f4b0d84cfc45cbf81356e?auth=e946fd4077294408c08e31076a5535d4#76">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068796">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2494daf130d746abb0c2239f3bad873b?auth=2935479aa50e3e5553a5ad6a565cb4bf#111">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5fd7b53ce96e41a699872a432741df3e?auth=d89692fad191504c2039b8d0bedd4ab5#63">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7771da723d474fa5ab8fe3322a9ed0d4?auth=1408dd65d16e830a5fbdd6303588415d#63">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068797">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/2494daf130d746abb0c2239f3bad873b?auth=2935479aa50e3e5553a5ad6a565cb4bf#128">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5fd7b53ce96e41a699872a432741df3e?auth=d89692fad191504c2039b8d0bedd4ab5#79">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7771da723d474fa5ab8fe3322a9ed0d4?auth=1408dd65d16e830a5fbdd6303588415d#68">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestDeeplinkOneDeviceNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068795">test_public_chat_open_using_deep_link</a>, id: 702776 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cb84e1957e6844b38f854b04772cb1b5?auth=70ac118666a349e4e0492132bd77f79b#80">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068793">test_deep_link_with_invalid_user_public_key_own_profile_key</a>, id: 702774 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cb84e1957e6844b38f854b04772cb1b5?auth=70ac118666a349e4e0492132bd77f79b#41">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3068794">test_deep_link_open_user_profile</a>, id: 702775 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cb84e1957e6844b38f854b04772cb1b5?auth=70ac118666a349e4e0492132bd77f79b#98">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

VladimrLitvinenko commented:
hi @OmarBasem thanx for your work. The current PR is tested and ready to be merged. The https://github.com/status-im/status-mobile/issues/14613 issue is not reproducible anymore. Marked it as closed. One additional issue with low priority I have created as a follow-up https://github.com/status-im/status-mobile/issues/14624

flexsurfer commented:
@OmarBasem it seems like you reverted a few files in this PR

flexsurfer commented:
@OmarBasem I've reverted the commit, could you please open a new PR and fix reverted files , thanks

OmarBasem commented:
@flexsurfer sure!
